asab/zookeeper/container.py

```diff
@@ -56,9 +57,12 @@ def advertise(self, data, path):
 
 
 	async def _do_advertise(self, event_name):
-		for adv in self.Advertisments:
-			await adv._do_advertise(self)
-
+		try:
+			for adv in self.Advertisments:
+				await adv._do_advertise(self)
+		except aiozk.exc.NoNode:
+			L.exception("The path entry is missing ")
+			raise Exception("The path entry is missing")
```

Comment:
`raise RuntimeException("The path entry is missing")`

... also I don't think that this is a correct interpretation of that error; what do you mean by "The path entry is missing" ?

Reply:
@ateska the issue is node does not exist to set data.

My interpretation is because the path entry here

https://github.com/TeskaLabs/asab/blob/a37559ce78958f58d9580bef7e049773f7ab7029/asab/zookeeper/service.py#L23

failed to be created in zookeeper. 

Can I do
raise RuntimeException("Failed to write data to zookeeper because the node {} does not exist.".format(self.Path))
?

Reply:
That's better.

Reply:
@ateska any reason why we need to raise RuntimeException ?

Reply:
Well, you should not use "Exception" directly - it is generic type.
If unsure, use "RuntimeException" - but you can indeed find a better specific exception.